numpy/core/src/umath/scalarmath.c.src

```diff
@@ -190,12 +199,11 @@ static NPY_INLINE int
 
 static NPY_INLINE int
 @name@_ctype_remainder(@type@ a, @type@ b, @type@ *out) {
-    if (a == 0 || b == 0) {
-        *out = 0;
+    if (DIVIDEBYZERO_CHECK) {
         if (b == 0) {
-            return NPY_FPE_DIVIDEBYZERO;
+            *out = 0;
         }
-        return 0;
+        return NPY_FPE_DIVIDEBYZERO;
```

Comment:
Shouldn't the above be:
```
*out = 0;
if (b == 0) {
    return NPY_FPE_DIVIDEBYZERO;
}
return 0;
```
Since this is a remainder?  I am surprised the test wouldn't catch it though, so maybe I am misreading.

Other than this question, I think this is ready.  Thanks for working on fixing this long-standing issue!